json_typegen_shared/src/generation/rust.rs

```diff
@@ -48,23 +48,32 @@ pub fn rust_program(name: &str, shape: &Shape, options: Options) -> Code {
     }
 }
 
-pub fn rust_types(name: &str, shape: &Shape, options: Options) -> (Ident, Option<Code>) {
+pub fn rust_types(
+    name: &str,
+    shape: &Shape,
+    options: Options,
+) -> (Ident, Option<Code>) {
     let mut ctxt = Ctxt {
         options,
         type_names: HashSet::new(),
     };
 
-    type_from_shape(&mut ctxt, name, shape)
+    let (ident, code) = type_from_shape(&mut ctxt, name, shape);
+    alias(ident, name, code, &ctxt.options)
 }
 
-fn type_from_shape(ctxt: &mut Ctxt, path: &str, shape: &Shape) -> (Ident, Option<Code>) {
+fn type_from_shape(
+    ctxt: &mut Ctxt,
+    path: &str,
+    shape: &Shape,
+) -> (Ident, Option<Code>) {
     use crate::shape::Shape::*;
     match shape {
         Null | Any | Bottom => ("::serde_json::Value".into(), None),
         Bool => ("bool".into(), None),
         StringT => ("String".into(), None),
-        Integer => ("i64".into(), None),
-        Floating => ("f64".into(), None),
+        Integer => ("Decimal".into(), None),
```

Comment:
Seems straight-forward enough. Have you tested what happens when you try to call the macro from a different package?